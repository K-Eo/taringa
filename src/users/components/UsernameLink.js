import React, { PureComponent } from 'react'
import { Link } from 'react-router-dom'
import { connect } from 'react-redux'

export class UsernameLink extends PureComponent {
  render() {
    const { username, className } = this.props

    return (
      <Link className={className} to={`/u/${username}`}>
        {username}
      </Link>
    )
  }
}

const mapStateToProps = (state, ownProps) => {
  return state.entities.users[ownProps.id]
}

export default connect(mapStateToProps)(UsernameLink)
