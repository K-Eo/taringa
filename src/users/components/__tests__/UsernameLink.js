import React from 'react'

import { UsernameLink } from '../UsernameLink'

describe('Username Link', () => {
  it('exists', () => {
    expect(UsernameLink).toBeDefined()
  })

  it('renders default', () => {
    const wrapper = shallow(<UsernameLink />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders username', () => {
    const wrapper = shallow(<UsernameLink username="foo" />)

    expect(wrapper).toMatchSnapshot()
  })

  it('renders with addition class', () => {
    const wrapper = shallow(<UsernameLink className="bar" username="foo" />)

    expect(wrapper).toMatchSnapshot()
  })
})
