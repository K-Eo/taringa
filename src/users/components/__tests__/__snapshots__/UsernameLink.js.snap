// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Username Link renders default 1`] = `
<Link
  replace={false}
  to="/u/#USERNAME"
>
  #USERNAME
</Link>
`;

exports[`Username Link renders username 1`] = `
<Link
  replace={false}
  to="/u/foo"
>
  foo
</Link>
`;

exports[`Username Link renders with addition class 1`] = `
<Link
  className="bar"
  replace={false}
  to="/u/foo"
>
  foo
</Link>
`;
